### ISSUE TITLE
presto_server crashes when running TPC-H testing

### ISSUE THREAD

**Issue by jiangjiguang** [facebookincubator/velox#3800]
### Bug description

I am running TPC-H testing, but the presto_server crashes when executing different SQL(from No. 1 to No.22)
the Dataset is on Alibaba OSS connected by AWS.
No coredump, no error log when  presto_server crashed.

besides, presto_server.INFO refreshes every  about 4 seconds, how can I see the latest log？

### System information

presto version: 0.278
velox vesion: 2022-10-24 
java8
Ubuntu 20.04
AWS to connect Alibaba OSS
TPC-H(SF10)

### Relevant logs

```bash
Exception in thread "main" java.sql.SQLException: Query failed (#20230125_081118_00372_48btq): Encountered too many errors talking to a worker node. The node may have crashed or be under too much load. This is probably a transient issue, so please retry your query in a few minutes. (http://192.168.0.163:7777/v1/task/async/20230125_081118_00372_48btq.0.0.0/results/0/0 - 604 failures, failure duration 300.10s, total failed request time 300.43s)
        at com.facebook.presto.jdbc.PrestoResultSet.resultsException(PrestoResultSet.java:1841)
        at com.facebook.presto.jdbc.PrestoResultSet$ResultsPageIterator.computeNext(PrestoResultSet.java:1821)
        at com.facebook.presto.jdbc.PrestoResultSet$ResultsPageIterator.computeNext(PrestoResultSet.java:1760)
        at com.facebook.presto.jdbc.internal.guava.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:141)
        at com.facebook.presto.jdbc.internal.guava.collect.AbstractIterator.hasNext(AbstractIterator.java:136)
        at com.facebook.presto.jdbc.internal.guava.collect.TransformedIterator.hasNext(TransformedIterator.java:42)
        at com.facebook.presto.jdbc.internal.guava.collect.Iterators$ConcatenatedIterator.getTopMetaIterator(Iterators.java:1311)
        at com.facebook.presto.jdbc.internal.guava.collect.Iterators$ConcatenatedIterator.hasNext(Iterators.java:1327)
        at com.facebook.presto.jdbc.PrestoResultSet.next(PrestoResultSet.java:146)
        at com.pangpang.presto.client.jdbc.PrestoBenchmark.performanceItem(PrestoBenchmark.java:79)
        at com.pangpang.presto.client.jdbc.PrestoBenchmark.performance(PrestoBenchmark.java:59)
        at com.pangpang.presto.client.jdbc.PrestoBenchmark.main(PrestoBenchmark.java:46)
```

**Comment by aditi-pandit** [comment]
@majetideepak 
We had some internal problems running with AWS/S3 combination internally at Ahana also. More at https://github.com/facebookincubator/velox/issues/3828

**Comment by majetideepak** [comment]
@jiangjiguang I have been seeing this on AWS S3 and just found a fix yesterday. Can you apply the following patch and verify if it fixes the issue?
```
diff --git a/velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp b/velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp
index a7905869..b79bac5c 100644
--- a/velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp
+++ b/velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp
@@ -233,6 +233,7 @@ class S3FileSystem::Impl {
     if (origCount == 0) {
       Aws::SDKOptions awsOptions;
       awsOptions.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Fatal;
+      awsOptions.httpOptions.installSigPipeHandler = true;
       Aws::InitAPI(awsOptions);
     }
   }
```

**Comment by jiangjiguang** [comment]
@majetideepak OK, I will verify it.

**Comment by majetideepak** [comment]
@jiangjiguang are you able to verify this fix?
We have a PR up here https://github.com/facebookincubator/velox/pull/3843

**Comment by jiangjiguang** [comment]
> 

Sorry for delay, I will verify it today.

**Comment by jiangjiguang** [comment]
> @jiangjiguang are you able to verify this fix? We have a PR up here #3843

@majetideepak The PR is good, and presto_server dose not crash. Thanks

**Comment by majetideepak** [comment]
Resolved with https://github.com/facebookincubator/velox/pull/3843
